src/ShallowWaterRosenbrock.jl

```diff
@@ -0,0 +1,243 @@
+function shallow_water_rosenbrock_time_step!(
+     y₂, ϕ, F, q₁, q₂, duh₁, duh₂, H1h, H1hchol, y_wrk,  # in/out args
```

Comment:
If I am not wrong, `y2` represents `y^{n+1}` in the draft paper, and `y1` represents  `y^{n}`, right? What does `ym` represent in the draft paper? As far as I can see, the calling function is providing `y^{n-1}` as an argument to the `ym` function parameter. Is that we we actually want/what it is written in the draft paper? Can we use better parameter names to reflect what `y2` `y1` and `ym` represent using draft notation?

Reply:
Personally I am OK with having these as `y2`, `y1`, `ym`, but if you prefer that they be something like `yp2`, `yp1`, `y0` or something like that I can change them. I just don't like having too many underscore letters in a row

Reply:
Sounds good, done :)